src/index/updater.rs

```diff
@@ -179,13 +179,13 @@ impl Updater {
     index: &Index,
     mut height: u64,
     index_sats: bool,
+    options: Options,
   ) -> Result<mpsc::Receiver<BlockData>> {
     let (tx, rx) = mpsc::sync_channel(32);
 
     let height_limit = index.height_limit;
 
-    let client =
-      Client::new(&index.rpc_url, index.auth.clone()).context("failed to connect to RPC URL")?;
+    let client = options.bitcoin_rpc_client()?;
```

Comment:
Why not just `let client = index.options.bitcoin_rpc_client()?;`, and then we don't have to pass the cloned `options` in here at all?

Reply:
jup

Reply:
jup